tokio/src/io/poll_evented.rs

```diff
@@ -126,6 +126,7 @@ impl<E: Source> PollEvented<E> {
 
     /// Deregisters the inner io from the registration and returns a Result containing the inner io.
     #[cfg(any(feature = "net", feature = "process"))]
+    #[cfg(not(target_os = "wasi"))]
     pub(crate) fn into_inner(mut self) -> io::Result<E> {
```

Comment:
Why don't these methods exist on WASM?

Reply:
We can't poll events on Wasi since threading it's available.

Reply:
@rjzak You can poll. You just cannot wake up another thread since there are no threads (yet).

Reply:
This was fixed